### PR TITLE
Calculate and Store site-level QCs

### PR DESCRIPTION
**Pull request by kcibul**
Addresses [219](https://github.com/broadinstitute/dsp-spec-ops/issues/219)

Major changes

- calculate site level metrics in `feature_extract.sql`
- extract metrics, apply thresholds, and set filter field in ExtractFeature
- CreateSiteFilteringFiles to translate from input VCF with filter fields into format for BQ loading, especially `location` fields
- update WDL to call CreateSiteFilteringFiles and upload results to BQ

Minor changes
 - added call_GQ to alt_allele creation
 - reduced memory requirements in WDL

### CODE REVIEW

File: scripts/variantstore/wdl/ngs_filter_extract.wdl [hunk 1]
```diff
@@ -154,7 +160,8 @@ workflow NgsFilterExtract {
         indel_recal_tranches = IndelsVariantRecalibrator.tranches,
 
         fq_info_destination_table = fq_filter_set_info_table,
-        fq_tranches_destination_table = fq_filter_set_tranches_table
+        fq_tranches_destination_table = fq_filter_set_tranches_table,
```
Comment (mmorgantaylor) [inline scripts/variantstore/wdl/ngs_filter_extract.wdl:163]: i know this isn't a change in your PR, but while we're in here, could we change `fq_filter_set_tranches_table` to `fq_tranches_destination_table` everywhere? that would clarify things so much for me (and +1 for consistency)

File: scripts/variantstore/wdl/ngs_filter_extract.wdl [hunk 2]
```diff
@@ -328,22 +339,29 @@ task UploadFilterSetToBQ {
...
+        gatk --java-options "-Xmx1g" \
             CreateFilteringFiles \
             --ref-version 38 \
             --filter-set-name ~{filter_set_name} \
             -mode SNP \
             -V ~{snp_recal_file} \
             -O ~{filter_set_name}.snps.recal.tsv
 
-        gatk --java-options "-Xmx4g" \
+        gatk --java-options "-Xmx1g" \
```
Comment (mmorgantaylor) [inline scripts/variantstore/wdl/ngs_filter_extract.wdl:350]: nice catch!

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/CreateSiteFilteringFiles.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import java.io.IOException;
+
+@CommandLineProgramProperties(
+        summary = "Create files for ingesting Site-level filtering data into BigQuery",
+        oneLineSummary = "Site-level filtering Data Ingest tool for GVS",
+        programGroup = ShortVariantDiscoveryProgramGroup.class,
+        omitFromCommandLine = true
+)
+public final class CreateSiteFilteringFiles extends VariantWalker {
+    static final Logger logger = LogManager.getLogger(CreateVariantIngestFiles.class);
```
Comment (mmorgantaylor) [inline src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/CreateSiteFilteringFiles.java:32]: should this use CreateSiteFilteringFiles.class instead?

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/CreateSiteFilteringFiles.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+    private SimpleXSVWriter writer;
+
+    private List<String> HEADER =
```
Comment (mmorgantaylor) [inline src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/CreateSiteFilteringFiles.java:36]: for consistency with other tools, move this to SchemaUtils ?

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/ExtractFeaturesEngine.java [hunk 5]
```diff
@@ -234,6 +252,14 @@ private void processVQSRRecordForPosition(ExtractFeaturesRecord rec) {
         }
         builder.attribute(GATKVCFConstants.EXCESS_HET_KEY, String.format("%.3f", excessHetApprox));
 
+        if (rec.getDistinctAlleles() > 6) {
```
Comment (mmorgantaylor) [inline src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/ExtractFeaturesEngine.java:255]: is this a constant defined somewhere that we can use? (or should we make it one?)

File: src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVCFHeaderLines.java [hunk 6]
```diff
@@ -75,6 +75,9 @@ public static VCFFormatHeaderLine getEquivalentFormatHeaderLine(final String inf
         addFilterLine(new VCFFilterHeaderLine(VCFConstants.PASSES_FILTERS_v4, "Site contains at least one allele that passes filters"));
 
         addFilterLine(new VCFFilterHeaderLine(NAY_FROM_YNG, "Considered a NAY in the Yay, Nay, Grey table"));
+        addFilterLine(new VCFFilterHeaderLine(EXCESS_ALLELES, "Site has an excess of alternate alleles"));
```
Comment (mmorgantaylor) [inline src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVCFHeaderLines.java:78]: if the excess alleles number (6) was a constant somewhere, you could put that into the header. (maybe users don't care, but i would personally find it helpful!)

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/ExtractFeaturesRecord.java [hunk 7]
```diff
@@ -1,12 +1,9 @@
 package org.broadinstitute.hellbender.tools.variantdb.nextgen;
 
 import htsjdk.samtools.util.Locatable;
-import org.apache.avro.generic.GenericData;
 import org.apache.avro.generic.GenericRecord;
 import org.broadinstitute.hellbender.tools.variantdb.SchemaUtils;
 
-import java.util.ArrayList;
-import java.util.HashMap;
```
Comment (mmorgantaylor) [inline src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/ExtractFeaturesRecord.java:9]: thanks for the cleanup!

**Review by mmorgantaylor — APPROVED** [review]
looks great, only minor comments/questions

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/CreateSiteFilteringFiles.java (hunk 3 above)
Comment (kcibul) [reply src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/CreateSiteFilteringFiles.java:32]: yup

**Pull request merged**